create_shards.c

```diff
@@ -534,3 +578,40 @@ IntegerToText(int32 value)
 
 	return valueText;
 }
+
+
+/*
+ *	SupportFunctionForColumn helps to find the support function given a column, an access
+ *	method and id of a support function. This function returns InvalidOid if there is no
+ *	support function associated with the data type of the column. Also, this function
+ *	errors-out if there is no default operator class for the data type of the column.
+ */
+Oid
+SupportFunctionForColumn(Var *partitionColumn, Oid accessMethodId,
+		                 int16 supportFunctionNumber)
+{
+	Oid operatorFamilyId = InvalidOid;
+	Oid supportFunctionOid = InvalidOid;
+	Oid columnOid = partitionColumn->vartype;
+	Oid operatorClassId = GetDefaultOpClass(columnOid, accessMethodId);
+
+	/*
```

Comment:
This comment is actually incorrect, I think… [This documentation section](http://www.postgresql.org/docs/9.4/static/indexes-opclass.html) shows that an index can be defined using a customer operator class rather than the default. So it's possible we could support types without default operator classes, so long as the user specified which custom operator class to use at the time of distribution. I'll update this comment to clarify that we just don't support anything other than using a default operator class at the moment.